### PR TITLE
fix(agw): refactor QoSManager

### LINKED ISSUE #11242: pipelined: qos: get rid of get_qos_manager() from qos manager

### PR DESCRIPTION
**Pull request by ankitMeta1**
## Summary

Addressing issue(https://github.com/magma/magma/issues/11242) to refactor QosManager, to create a single instance in service_manager, clean the static function and variables from the class, and pass the QosManager instance  as context to apps.

## Test Plan
ran unit test locally(make test_python)
ran make test_python_service too, but its taking longer on local machine, hoping github cli would be faster.

### CODE REVIEW

**Review by koolzz — COMMENTED** [review]
not a big fan of initializing the same object 3 times, 

>>> ran make test_python_service too, but its taking longer on local machine, hoping github cli would be faster.

just as a heads up, ci doens't run the whole pipelined unit test suite so need to run that locally unfortunately

**Comment by ankitMeta1** [comment]
> not a big fan of initializing the same object 3 times,
> 
> > > > ran make test_python_service too, but its taking longer on local machine, hoping github cli would be faster.
> 
> just as a heads up, ci doens't run the whole pipelined unit test suite so need to run that locally unfortunately

not sure, if i understand the comment, which object are we initializing 3 times?

**Comment by koolzz** [comment]
> > not a big fan of initializing the same object 3 times,
> > > > > ran make test_python_service too, but its taking longer on local machine, hoping github cli would be faster.
> > 
> > 
> > just as a heads up, ci doens't run the whole pipelined unit test suite so need to run that locally unfortunately
> 
> not sure, if i understand the comment, which object are we initializing 3 times?

ah 2 times I suppose its not in enf_stats. `self._qos_mgr.init_impl(datapath)` but thats probably not worth a larger refactor

**Review by koolzz — APPROVED** [review]

File: lte/gateway/python/magma/pipelined/qos/common.py [hunk 1]
```diff
@@ -489,7 +494,7 @@ def add_subscriber_qos(
 
     def remove_subscriber_qos(self, imsi: str = "", del_rule_num: int = -1):
         with QosManager.lock:
-            if not self._qos_enabled or not self._initialized:
+            if not self._qos_enabled or not self._initialized or not self._is_impl_initialized():
```
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/qos/common.py:497]: @ankitbhadoria21, we dont need separate _is_impl_initialized(), can you post another PR to use _initialized to keep track of the initialization.

**Pull request merged**